tests/end-to-end/compass_e2e_test.go

```diff
@@ -1,14 +1,993 @@
+		Annotations: &graphql.Annotations{placeholder: placeholder},
+	}
+}
+
+func deleteApplication(t *testing.T, id string) {
+	req := gcli.NewRequest(fmt.Sprintf(`mutation {
+		deleteApplication(id: "%s") {
+			id
+		}	
+	}`, id))
+	require.NoError(t, tc.cli.Run(context.Background(), req, nil))
+}
+
+func deleteRuntime(t *testing.T, id string) {
+	delReq := gcli.NewRequest(fmt.Sprintf("mutation{deleteRuntime(id: \"%s\")  {id} }", id))
+	err := tc.cli.Run(context.Background(), delReq, nil)
+	require.NoError(t, err)
+}
+
+func fixBasicAuth() *graphql.AuthInput {
+	return &graphql.AuthInput{
+		Credential: &graphql.CredentialDataInput{
+			Basic: &graphql.BasicCredentialDataInput{
+				Username: "admin",
+				Password: "secret",
+			},
+		},
+		AdditionalHeaders: &graphql.HttpHeaders{
+			"headerA": []string{"ha1", "ha2"},
+			"headerB": []string{"hb1", "hb2"},
+		},
+		AdditionalQueryParams: &graphql.QueryParams{
+			"qA": []string{"qa1", "qa2"},
+			"qB": []string{"qb1", "qb2"},
+		},
+	}
+}
+
+func fixOauthAuth() *graphql.AuthInput {
+	return &graphql.AuthInput{
+		Credential: &graphql.CredentialDataInput{
+			Oauth: &graphql.OAuthCredentialDataInput{
+				URL:          "http://oauth/token",
+				ClientID:     "clientID",
+				ClientSecret: "clientSecret",
+			},
+		},
+	}
+}
+
+func fixDepracatedVersion1() *graphql.VersionInput {
+	return &graphql.VersionInput{
+		Value:           "v1",
+		Deprecated:      ptrBool(true),
+		ForRemoval:      ptrBool(true),
+		DeprecatedSince: ptrString("v5"),
+	}
+}
+
+// TODO: to test:
+// - cannot specify basic and auth at the same time
+// specify label created-by
+
+// testContext contains dependencies that help executing tests
+type testContext struct {
+	graphqlizer       graphqlizer
+	gqlFieldsProvider gqlFieldsProvider
+	cli               *gcli.Client
+}
+
+func getDirectorURL() string {
+	url := os.Getenv("DIRECTOR_GRAPHQL_API")
+	if url == "" {
+		url = "http://127.0.0.1:3000/graphql"
+	}
+	return url
+}
+
+// resultMapperFor returns generic object that can be passed to Run method for storing response.
+// In GraphQL, set `result` alias for your query
+func resultMapperFor(target interface{}) genericGQLResponse {
+	if reflect.ValueOf(target).Kind() != reflect.Ptr {
+		panic("target has to be a pointer")
+	}
+	return genericGQLResponse{
+		Result: target,
+	}
+}
+
+type genericGQLResponse struct {
+	Result interface{} `json:"result"`
+}
+
+func saveQueryInExamples(t *testing.T, query string, exampleName string) {
+	t.Helper()
+	sanitizedName := strings.Replace(exampleName, " ", "-", -1)
+	// replace uuids with constant value
+	r, err := regexp.Compile("[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}")
+	require.NoError(t, err)
+	query = r.ReplaceAllString(query, "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa")
+	err = ioutil.WriteFile(fmt.Sprintf("%s/src/github.com/kyma-incubator/compass/examples/%s.graphql", os.Getenv("GOPATH"), sanitizedName), []byte(query), 0660)
+	require.NoError(t, err)
```

Comment:
This file is too long. Please split it to separate files in a different package.

Reply:
Consider similar structure to the director project: `internal/domain` package with different domains, where there is a file per single test. Then it will be easier to put test parts into helper functions, and then they will be much cleaner.

Reply:
I created director package and split tests into 2 files: for app and runtimes